File: QRok/Models/Survey.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace QRok.Models
{
    public class Survey
    {
        public int Id { get; set; }

        public Guid Guid { get; set; }

        [Required]
        [MaxLength(255)]
        public string Title { get; set; }

        public IList<SurveyOption> SurveyOptions { get; set; }

        public DateTime CloseDateTime { get; set; }
        public DateTime DeleteDateTime { get; set; }

        public Survey()
        {
            SurveyOptions = new List<SurveyOption>();
        }
    }
}

File: QRok/wwwroot/js/KOViewModels/Survey.js
function Survey(obj) {
	this.Id = ko.observable(0);
	this.Guid = ko.observable("");
	this.Title = ko.observable("");
	this.CloseDateTime = ko.observable(null);
	this.DeleteDateTime = ko.observable(null);
	this.SurveyOptions = ko.observableArray();

	if (obj != null) {
		this.Id(obj.Id);
		this.Guid(obj.Guid);
		this.Title(obj.Title);
		this.CloseDateTime(obj.CloseDateTime);
		this.DeleteDateTime(obj.DeleteDateTime);
		for (var i = 0; i < obj.SurveyOptions.length; i++) {
			console.log(obj.SurveyOptions[i]);
			this.SurveyOptions.push(new SurveyOption(obj.SurveyOptions[i]));
		}
	}
	//console.log(ko.toJS(this));
}